New tool to detect outdated versions // and update GitHub managed ones

fulldecent opened the pull request:
# Summary

I used data from the [check-outdateds-actions](https://github.com/actions/starter-workflows/pull/3127) for the research to do this work.

This pull request carefully bumps the version of official CI GitHub Actions to a newer or latest version. 

This helps everyone because it will make it easier for us all to use the new runners when available. And it will help us see the new GitHub Actions and their features, rather than starting out a few versions behind.

- ✅ If the new or latest version supports the same syntax as the current usage, then the upgrade is in scope for this PR.
- ✅ If this repo currently pins a version because it wants a specific feature, but that was later introduced in a public maintained release of the updated action then updating that action is in scope for this PR.
- ❌ Any changes that require major refactoring because of changes in a GitHub Action are out of scope for this PR. But we should do that work later as a separate initiative. (I have not identified any specific cases of this.)
- ❌ Any changes to action versions other than `actions/...` are OUT of scope for this PR.
- ⚠️ Any formatting changes are out of scope of this PR. If I was already touching some part I may have fix formatting in just that part. But if this is unwelcome please tell me or send a recommendation commit and I can pull that in.


---

<!--
IMPORTANT:

This repository contains configuration for what users see when they click on the `Actions` tab and the setup page for Code Scanning.

It is not:
* A playground to try out scripts
* A place for you to create a workflow for your repository
-->

## Pre-requisites

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

^^ Not applicable. This is not a new workflow.

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least privileged [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _CI_ workflows, the workflow:**

- [x] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [x] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [x] Packaging workflows should run on `release` with `types: [ created ]`.
- [x] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `creator`: Name of the organization/user producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

fulldecent commented:
I have fixed the merge conflicts with the main branch and this is now clean again to merge

ScottBrenner commented on .github/workflows/sync-ghes.yaml:
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
+    - uses: actions/checkout@v5
```
```suggestion
    - uses: actions/checkout@v6
```

fulldecent replied:
I agree.

At the moment [I'm having trouble](https://github.com/microsoft/vscode/issues/169930) adding that update (across all occurrances) this this PR.

If anybody would commit access on this revoke can push that commit to my PR I would appreciate it

fulldecent committed:
Bump actions/ai-inference from v1 to v2

Breaking changes reviewed:
- Requires Actions Runner v2.327.1+ (Node 24 runtime)
- No API/workflow syntax changes required

Files updated: 1
- automation/summary.yml

Refs: https://github.com/actions/ai-inference/releases/tag/v2.0.0

fulldecent committed:
Bump actions/cache from v3 to v4

Breaking changes reviewed:
- Updated to Node 20 runtime
- New save-always flag added (optional, backward compatible)
- No workflow syntax changes required

Files updated: 7
- ci/php.yml
- ci/elixir.yml
- ci/haskell.yml
- ci/symfony.yml
- code-scanning/codescan.yml
- deployments/azure-webapps-php.yml
- deployments/azure-webapps-dotnet-core.yml

Refs: https://github.com/actions/cache/releases/tag/v4.0.0

fulldecent committed:
Bump actions/checkout from v4 to v6

Breaking changes reviewed:
- Credential storage moved from .git/config to $RUNNER_TEMP
- Requires Actions Runner v2.329.0+ for Docker container actions
- No workflow syntax changes required

Files updated: 174 total
- 170 major-only version references (@v4 → @v6)
- 4 SHA-pinned references with updated commits:
  * scorecard.yml: v4.2.2 → v6.0.1 (patch-level)
  * google-cloudrun-docker.yml: v4 → v6 (major-level)
  * google.yml: v4 → v6 (major-level)
  * google-cloudrun-source.yml: v4 → v6 (major-level)

Refs: https://github.com/actions/checkout/releases/tag/v6.0.0

fulldecent committed:
Bump actions/download-artifact from v4 to v6

Breaking changes reviewed:
- v5: Path behavior fixed for artifact downloads by ID (not used in these workflows)
- v6: Node 24 support added
- No workflow syntax changes required

Files updated: 7
- ci/python-publish.yml
- deployments/azure-webapps-java-jar-gradle.yml
- deployments/azure-webapps-php.yml
- deployments/azure-webapps-node.yml
- deployments/azure-webapps-python.yml
- deployments/azure-webapps-java-jar.yml
- deployments/azure-webapps-dotnet-core.yml

Refs: https://github.com/actions/download-artifact/releases/tag/v6.0.0

fulldecent committed:
Bump actions/first-interaction from v1 to v3

Breaking changes reviewed:
- v2.0.0: Converted from Container to TypeScript action (no workflow changes required)
- v3.0.0: Updated to Node 24 runtime

Files updated:
- automation/greetings.yml

Refs:
- https://github.com/actions/first-interaction/releases/tag/v2.0.0
- https://github.com/actions/first-interaction/releases/tag/v3.0.0

fulldecent committed:
Bump actions/github-script from v6 to v8

Breaking changes reviewed:
- v7.0.0: Updated to Node 20 runtime, added base-url option
- v8.0.0: Updated to Node 24 runtime, requires runner v2.327.1+

Files updated:
- deployments/openshift.yml

Refs:
- https://github.com/actions/github-script/releases/tag/v7.0.0
- https://github.com/actions/github-script/releases/tag/v8.0.0

fulldecent committed:
Bump actions/labeler from v4 to v6

Breaking changes reviewed:
- v5.0.0: Configuration file structure redesigned (only affects .github/labeler.yml user config, not workflow), Node 20 runtime, dot input default true, sync-labels bug fixed
- v6.0.0: Updated to Node 24 runtime, requires runner v2.327.1+

Files updated:
- automation/label.yml

Refs:
- https://github.com/actions/labeler/releases/tag/v5.0.0
- https://github.com/actions/labeler/releases/tag/v6.0.0

fulldecent committed:
Bump actions/setup-dotnet from v4 to v5

Breaking changes reviewed:
- Updated to Node 24 runtime, requires runner v2.327.1+
- Removed support for older .NET versions

Files updated:
- ci/dotnet.yml
- ci/dotnet-desktop.yml

Refs:
- https://github.com/actions/setup-dotnet/releases/tag/v5.0.0

fulldecent committed:
Bump actions/setup-go from v4 to v6

Breaking changes reviewed:
- v6.0.0: Updated to Node 24 runtime, requires runner v2.327.1+, improved toolchain handling

Files updated:
- ci/go.yml

Refs:
- https://github.com/actions/setup-go/releases/tag/v6.0.0

fulldecent committed:
Bump actions/setup-java from v4 to v5

Breaking changes reviewed:
- Updated to Node 24 runtime, requires runner v2.327.1+

Files updated:
- ci/android.yml, ci/ant.yml, ci/clojure.yml, ci/gradle.yml, ci/gradle-publish.yml, ci/maven.yml, ci/maven-publish.yml, ci/scala.yml
- code-scanning/codescan.yml
- deployments/azure-container-webapp.yml, deployments/azure-webapps-java-jar.yml, deployments/google-cloudrun-docker.yml, deployments/google-cloudrun-source.yml
- pages/jekyll-docker.yml, pages/jekyll-gh-pages.yml, pages/jekyll.yml

Refs:
- https://github.com/actions/setup-java/releases/tag/v5.0.0

fulldecent committed:
Bump actions/setup-node from v4 to v6

Breaking changes reviewed:
- Automatic caching now limited to npm only (previously included yarn and pnpm)

Files updated:
- ci/node.js.yml, ci/npm-publish.yml, ci/npm-publish-github-packages.yml, ci/webpack.yml, ci/laravel.yml, ci/rubyonrails.yml, ci/symfony.yml
- code-scanning/endorlabs.yml, code-scanning/jfrog-sast.yml, code-scanning/nowsecure-mobile-sbom.yml
- deployments/azure-webapps-node.yml, deployments/google-cloudrun-docker.yml, deployments/google-cloudrun-source.yml
- pages/gatsby.yml, pages/nextjs.yml, pages/nuxtjs.yml

Refs:
- https://github.com/actions/setup-node/releases/tag/v6.0.0

fulldecent committed:
Bump actions/setup-python from v3/v4/v5 to v6

Breaking changes reviewed:
- v6.0.0: Updated to Node 24 runtime, requires runner v2.327.1+, added pip-version support

Files updated:
- ci/django.yml, ci/pylint.yml, ci/python-app.yml, ci/python-package.yml, ci/python-package-conda.yml, ci/python-publish.yml
- code-scanning/bandit.yml
- deployments/azure-webapps-python.yml
- pages/pelican.yml

Refs:
- https://github.com/actions/setup-python/releases/tag/v6.0.0

fulldecent committed:
Bump actions/stale from v5 to v10

Breaking changes reviewed:
- v6.0.0: Node 16 runtime
- v7.0.0: Exempt items no longer have stale label managed by action
- v8.0.0: Build now aborts on errors (previously continued)
- v9.0.0: Action is now stateful (maintains progress between runs), Node 20 runtime
- v10.0.0: Updated to Node 24 runtime, requires runner v2.327.1+

Files updated:
- automation/stale.yml

Refs:
- https://github.com/actions/stale/releases/tag/v6.0.0
- https://github.com/actions/stale/releases/tag/v7.0.0
- https://github.com/actions/stale/releases/tag/v8.0.0
- https://github.com/actions/stale/releases/tag/v9.0.0
- https://github.com/actions/stale/releases/tag/v10.0.0

fulldecent committed:
Bump actions/upload-artifact from v4 to v6

Breaking changes reviewed:
- v5.0.0: Preliminary Node 24 support
- v6.0.0: Node 24 runtime by default, requires runner v2.327.1+

Files updated:
- code-scanning/anchore-syft.yml, code-scanning/clj-watson.yml, code-scanning/codescan.yml, code-scanning/dependency-review.yml, code-scanning/hadolint.yml, code-scanning/policy-validator-cfn.yaml
- pages/pelican.yml

Refs:
- https://github.com/actions/upload-artifact/releases/tag/v5.0.0
- https://github.com/actions/upload-artifact/releases/tag/v6.0.0

fulldecent committed:
Bump actions/download-artifact from v6 to v7

Breaking changes reviewed:
- v7.0.0: Node 24 runtime by default, requires runner v2.327.1+

Files updated:
- code-scanning/anchore-syft.yml, code-scanning/clj-watson.yml, code-scanning/codescan.yml, code-scanning/dependency-review.yml, code-scanning/hadolint.yml, code-scanning/policy-validator-cfn.yaml
- pages/pelican.yml

Refs:
- https://github.com/actions/download-artifact/releases/tag/v7.0.0

fulldecent committed:
Bump actions/cache from v4 to v5

Breaking changes reviewed:
- v5.0.0: Node 24 runtime, requires runner v2.327.1+

Files updated:
- ci/php.yml, ci/elixir.yml, ci/haskell.yml, ci/symfony.yml
- code-scanning/codescan.yml
- deployments/azure-webapps-php.yml, deployments/azure-webapps-dotnet-core.yml

Refs:
- https://github.com/actions/cache/releases/tag/v5.0.0

fulldecent committed:
Bump remaining actions to latest patch versions

Updates:
- actions/setup-python v6.0.0 → v6.1.0 (deployments/azure-webapps-python.yml)
- actions/upload-artifact SHA → v6.0.0 (code-scanning/scorecard.yml)
- actions/upload-pages-artifact v3 → v4 (9 pages workflows)

Breaking changes reviewed:
- upload-pages-artifact v4.0.0: Hidden files (dotfiles) no longer included by default

Files updated:
- deployments/azure-webapps-python.yml
- code-scanning/scorecard.yml
- pages/mdbook.yml, pages/nuxtjs.yml, pages/hugo.yml, pages/astro.yml, pages/gatsby.yml, pages/nextjs.yml, pages/jekyll-gh-pages.yml, pages/static.yml, pages/jekyll.yml

Refs:
- https://github.com/actions/setup-python/releases/tag/v6.1.0
- https://github.com/actions/upload-artifact/releases/tag/v6.0.0
- https://github.com/actions/upload-pages-artifact/releases/tag/v4.0.0

fulldecent committed:
Fix missed setup-dotnet v4 references

Files updated:
- code-scanning/defender-for-devops.yml
- code-scanning/ossar.yml (commented reference)
- deployments/azure-functions-app-dotnet.yml
- deployments/azure-webapps-dotnet-core.yml

fulldecent committed:
Fix remaining cache v4 references in pages workflows

Files updated:
- pages/nuxtjs.yml
- pages/gatsby.yml
- pages/nextjs.yml

fulldecent committed:
Update final checkout SHA references to v6.0.1

Files updated:
- code-scanning/policy-validator-cfn.yaml
- code-scanning/policy-validator-tf.yaml

Refs:
- https://github.com/actions/checkout/releases/tag/v6.0.1

fulldecent commented:
Thank you for taking a look. I don't know if this repo is accepting contributions. But ping me if somebody here has commit access and needs a fresh up on this PR.

edgarcalderon171101-rgb commented:
Hey ofc all of are pubic repos are open for community add once if you would like me too make one for a spasific topi lmk I have a lot of projects and the public once are juts things I can show

edgarcalderon171101-rgb reviewed (changes requested):
These seemed to be a lot of errors and I don't want too risk it being bad code that can be harmful
